clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -535,68 +441,90 @@ raw2trace_t::merge_and_process_thread_files()
             offline_entry_t entry;
             if (!thread_files[tidx]->read((char*)&entry, sizeof(entry)) ||
                 entry.addr.type != OFFLINE_TYPE_IFLUSH)
-                FATAL_ERROR("Flush missing 2nd entry");
+                return "Flush missing 2nd entry";
             VPRINT(2, "Flush " PFX"-" PFX"\n", (ptr_uint_t)in_entry.addr.addr,
                    (ptr_uint_t)entry.addr.addr);
             size += instru.append_iflush(buf, in_entry.addr.addr,
                                          (size_t)(entry.addr.addr - in_entry.addr.addr));
             buf += size;
-        } else
-            FATAL_ERROR("Unknown trace type %d", (int)in_entry.timestamp.type);
+        } else {
+            std::stringstream ss;
+            ss << "Unknown trace type " << (int)in_entry.timestamp.type;
+            return ss.str();
+        }
         if (size > 0) {
             CHECK((uint)size < MAX_COMBINED_ENTRIES, "Too many entries");
-            if (!out_file.write((char*)buf_base, size))
-                FATAL_ERROR("Failed to write to output file");
+            if (!out_file->write((char*)buf_base, size))
+                return "Failed to write to output file";
         }
     } while (thread_count > 0);
+    return std::string();
```

Comment:
Sometimes you're using 'return ""' and others 'return std::string()' -- best to pick one.  IMHO "" is clearer.